draft-ietf-quic-recovery.md

```diff
@@ -280,23 +296,26 @@ connection's final smoothed RTT value as the resumed connection's initial RTT.
 If no previous RTT is available, or if the network changes, the initial RTT
 SHOULD be set to 100ms.
 
-When a handshake packet is sent, the sender SHOULD set an alarm for the
-handshake timeout period.
+When CRYPTO_HS frames are sent, the sender SHOULD set an alarm for the handshake
+timeout period.
 
-When the alarm fires, the sender MUST retransmit all unacknowledged handshake
-data, by calling RetransmitAllUnackedHandshakeData(). On each consecutive
-firing of the handshake alarm, the sender SHOULD double the handshake timeout
-and set an alarm for this period.
+When the alarm fires, the sender MUST retransmit all unacknowledged CRYPTO_HS
+data by calling RetransmitAllUnackedHandshakeData(). On each
+consecutive firing of the handshake alarm without receiving an
+acknowledgement for a new packet, the sender SHOULD double the handshake
+timeout and set an alarm for this period.
 
 When an acknowledgement is received for a handshake packet, the new RTT is
 computed and the alarm SHOULD be set for twice the newly computed smoothed RTT.
 
-Handshake data may be cancelled by handshake state transitions. In particular,
-all non-protected data SHOULD no longer be transmitted once packet protection
-is available.
+#### Retry
+
+A RETRY packet causes the content of the client's Initial packet to be
+immediately retransmitted along with the token present in the RETRY.
 
-(TODO: Work this section some more. Add text on client vs. server, and on
-stateless retry.)
+The RETRY indicates that the Initial was received but not processed.
+It MUST NOT be treated as an acknowledgment for the Initial,
+but it MAY be used for an RTT measurement.
```

Comment:
I think this might limit what a load balancer can do.  I'm thinking that, if the stateless retry ends up in a different server, the RTT measurement may be wrong.  Not a big deal, but I don't see why the 'MAY' needs to be included.